ale/base/data_naif.py

```diff
@@ -337,11 +338,36 @@ def sensor_position(self):
                 # spkezr returns a vector from the observer's location to the aberration-corrected
                 # location of the target. For more information, see:
                 # https://naif.jpl.nasa.gov/pub/naif/toolkit_docs/FORTRAN/spicelib/spkezr.html
-                state, _ = spice.spkezr(target,
-                                        time,
-                                        self.reference_frame,
-                                        self.light_time_correction,
-                                        observer)
+                if self.correct_lt_to_surface and self.light_time_correction.upper() == 'LT+S':
+                    obs_tar_state, obs_tar_lt = spice.spkezr(target,
+                                                             time,
+                                                             'J2000',
+                                                             self.light_time_correction,
+                                                             observer)
+                    # ssb to spacecraft
+                    ssb_obs_state, ssb_obs_lt = spice.spkezr(observer,
+                                                             time,
+                                                             'J2000',
+                                                             'NONE',
+                                                             'SSB')
+
+                    radius_lt = (self.target_body_radii[2] + self.target_body_radii[0]) / 2 / (scipy.constants.c/1000.0)
+                    adjusted_time = time - obs_tar_lt + radius_lt
+                    ssb_tar_state, ssb_tar_lt = spice.spkezr(target,
+                                                             adjusted_time,
+                                                             'J2000',
+                                                             'NONE',
+                                                             'SSB')
+                    state = ssb_tar_state - ssb_obs_state
+                    matrix = spice.sxform("J2000", self.reference_frame, time)
```

Comment:
The above calculation did not work with self.reference_frame, rather than "J2000", so I needed to set "J2000" for the individual steps, and then rotate back to `self.reference_frame` at the end.

Reply:
Just to confirm -- I tried this and it didn't work.